### ISSUE TITLE
Python problem on OSX

### ISSUE THREAD

**Issue by dmiljkovic** [Microsoft/LightGBM#519]
Hello,

Python examples are producing Segmentation fault error.  

~/LightGBM/examples/python-guide (master)$ python simple_example.py
Load data...
Start training...
Segmentation fault: 11

I can run examples using lightgbm image. 
LightGBM/examples/binary_classification (master)$ ../../lightgbm config=train.conf

## Environment info
OSX Siera
g++ (MacPorts gcc47 4.7.4_5) 4.7.4
Python 2.7.12 :: Anaconda 2.4.1 (x86_64

**Comment by guolinke** [comment]
@dmiljkovic just try it. Both success in ubuntu/windows. I don't have a mac to test. 
Which version did you use? The latest master ?

**Comment by dmiljkovic** [comment]
I used latest master version. I was able to install on ubuntu. Having problem running test examples on OSX.
College of mine got same error on his mac.

**Comment by dmiljkovic** [comment]
Produces error on lgb.train in simple_example.py
# train
gbm = lgb.train(params,
                lgb_train,
                num_boost_round=20,
                valid_sets=lgb_eval,
                early_stopping_rounds=5)

**Comment by guolinke** [comment]
@dmiljkovic 
refer to #521 
Did you use gcc-7 ?

**Comment by dmiljkovic** [comment]
Yes. gcc-7 is used 

~/LightGBM/build (master)$ cmake ..
-- The C compiler identification is GNU 7.1.0
-- The CXX compiler identification is GNU 7.1.0
-- Checking whether C compiler has -isysroot
-- Checking whether C compiler has -isysroot - yes
-- Checking whether C compiler supports OSX deployment target flag
-- Checking whether C compiler supports OSX deployment target flag - yes
-- Check for working C compiler: /usr/local/bin/gcc-7
-- Check for working C compiler: /usr/local/bin/gcc-7 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking whether CXX compiler has -isysroot
-- Checking whether CXX compiler has -isysroot - yes
-- Checking whether CXX compiler supports OSX deployment target flag
-- Checking whether CXX compiler supports OSX deployment target flag - yes
-- Check for working CXX compiler: /usr/local/bin/g++-7
-- Check for working CXX compiler: /usr/local/bin/g++-7 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   _lightgbm

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/dejanm/LightGBM/build

**Comment by dmiljkovic** [comment]
Solved!!

in CMakeLists.txt change compiler compiler version from 7 to 6

 

if(APPLE)
    SET(CMAKE_CXX_COMPILER "g++-6")
    SET(CMAKE_C_COMPILER "gcc-6")
endif()

**Comment by TengGao** [comment]
I have followed the above steps, and edited the CMakeLists.txt. I got through the
make -j
without error. But, when I did
cd ../python-package
python setup.py install 
I got the following error message: 
Install lib_lightgbm from: ['../lib_lightgbm.so']
running install
running bdist_egg
running egg_info
writing requirements to lightgbm.egg-info/requires.txt
writing lightgbm.egg-info/PKG-INFO
writing top-level names to lightgbm.egg-info/top_level.txt
writing dependency_links to lightgbm.egg-info/dependency_links.txt
reading manifest file 'lightgbm.egg-info/SOURCES.txt'
writing manifest file 'lightgbm.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.10-x86_64/egg
running install_lib
running build_py
creating build/bdist.macosx-10.10-x86_64/egg
error: could not create 'build/bdist.macosx-10.10-x86_64/egg': Permission denied

I am using anaconda, so I git clone the lightgbm package within the anaconda/bin.
Can anyone help me?
